django/db/models/query.py

```diff
@@ -1276,40 +1329,70 @@ def db(self):
     # PRIVATE METHODS #
     ###################
 
-    def _insert(self, objs, fields, returning_fields=None, raw=False, using=None, ignore_conflicts=False):
+    def _insert(
+        self, objs, fields, returning_fields=None,
+        raw=False, using=None, on_conflicts=ON_CONFLICTS_NONE, update_fields=[],
+        unique_fields=[]
+    ):
         """
         Insert a new record for the given model. This provides an interface to
         the InsertQuery class and is how Model.save() is implemented.
         """
         self._for_write = True
         if using is None:
             using = self.db
-        query = sql.InsertQuery(self.model, ignore_conflicts=ignore_conflicts)
+        query = sql.InsertQuery(
+            self.model, on_conflicts=on_conflicts, update_fields=update_fields,
+            unique_fields=unique_fields
+        )
         query.insert_values(fields, objs, raw=raw)
         return query.get_compiler(using=using).execute_sql(returning_fields)
     _insert.alters_data = True
     _insert.queryset_only = False
 
-    def _batched_insert(self, objs, fields, batch_size, ignore_conflicts=False):
+    def _check_on_conflicts_supported(self, on_conflicts, unique_fields):
+        if on_conflicts == ON_CONFLICTS_IGNORE:
+            feature = 'supports_ignore_conflicts'
+        elif on_conflicts == ON_CONFLICTS_UPDATE:
+            if unique_fields:
+                feature = 'supports_update_conflicts_with_unique_fields'
+            else:
+                feature = 'supports_update_conflicts_without_unique_fields'
+        else:
+            feature = None
+        if feature and not getattr(connections[self.db].features, feature):
+            raise NotSupportedError(
+                'This database backend does not support %s conflicts.' % (feature)
+            )
```

Comment:
It would be more readable to raise an error explicitly (like previously), e.g.
```python
db_features = connections[self.db].features
if ignore_conflicts and not db_features.supports_ignore_conflicts:
    raise NotSupportedError('This database backend does not support ignoring conflicts.')
if update_conflicts:
    if not db_feature.supports_update_conflicts:
        raise NotSupportedError(
            'This database backend does not support updating conflicts.'
        )
    if unique_fields and not db_features.supports_update_conflicts_with_target:
        raise NotSupportedError(
            'This database backend does not support updating conflicts with '
            'specifying unique fields that will trigger the upsert.'
        )
```
(I used new feature flags.)